shardy/dialect/sdy/ir/attrs.td

```diff
@@ -738,6 +779,19 @@ def Sdy_OpShardingRule : AttrDef<Sdy_Dialect, "OpShardingRule"> {
     these ops, so a user must tell it how. When it is a custom rule, then the
     rule is always preserved/never removed. `is_custom_rule` can only be true
     for `stablehlo.custom_call` ops.
+
+    Constraints:
+
+    - Number of mappings and types match.
+    - There is at least one mapping (can't have a rule for an op with no
+      operands/results).
+    - Rank of each `TensorMappingAttr` matches the rank of the corresponding
+      tensor type.
+    - No duplicate factor indices in each `TensorMappingAttr`.
+    - Per `DimMappingAttr`:
+      * There is at least one factor index.
+      * Factor indices must be in range [0, `factorSizes.size()`).
```

Comment:
we should refer to the field $factor_sizes here

Reply:
Let me know if this is what you meant.